src/software/geom/shot.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "software/geom/point.h"
+
+class Shot
+{
+   public:
+    /**
+     * Creates a shot with the given target and angle
+     *
+     * @param point The target to shoot at
+     * @param angle The angle formed by the shot origin, and the edges of the two
+     * obstacles closest to the shot path
+     *
+     */
+    Shot(Point point, Angle angle);
+
+    /**
+     * Returns a point, the target to shoot at.
+     * @return a point, representing the target to shoot at.
+     */
+    Point getPointToShootAt() const;
+
+    /**
+     * Returns the angle formed by the shot origin, and the edges of the two obstacles
+     * closest to the shot path
+     * @return the angle formed by the shot origin, and the edges of the two obstacles
+     * closest to the shot path
+     */
+    Angle getOpenAngle() const;
+
+   private:
+    /**
+     * The target of the shot
+     *
+     */
+    Point point;
+
+    /**
+     * The angle formed by the shot origin, and the edges of the two obstacles closest to
+     the shot path
+     *
```

Comment:
This is awesome, super clear for anyone trying to understand what angle means :+1: